### PR TITLE
Change/ignore [iSTD] validation

### PR DESCRIPTION
**Pull request by biodavidjm**
This pull request introduces improvements to the metabolomics RefMet name validation logic, enhances user-facing documentation, and adds robust tests for new behaviors. The most important changes are grouped below:

### RefMet Name Validation Improvements

* The `validate_refmetname` function now **skips internal standards** (`[iSTD]`) during RefMet name validation instead of stripping and validating them, ensuring more accurate reporting and less noise in validation output. [[1]](diffhunk://#diff-ff467faf48a54f61eaba9aced37a9097ca3ab1a4b4682e51fe7a49c2be843b43L89-R156) [[2]](diffhunk://#diff-51920e95310ebfbc1ae31709f3b95f89afffbf4f1a6e38e8b2b406e2fb6197eaR2-R5)
* The function now **accepts alternative names separated by slashes** (e.g., `"Leucine/Isoleucine"`) if any component matches the RefMet standard, improving flexibility for common naming conventions.
* Input validation is strengthened: the function now **requires a data frame with a `refmet_name` column** and errors clearly for invalid input, preventing silent failures.

### User Documentation and Messaging

* The `README.md` is improved with a clearer overview, feature list, installation instructions, and citation information, making it easier for new users to get started and cite the package. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L17-R52) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L58-R80)
* All user-facing validation messages are clarified and standardized for better readability and context during QC runs.

### Testing and Maintenance

* New tests are added to ensure that `[iSTD]` entries are skipped, slash-containing names are accepted, and invalid inputs are properly handled, increasing confidence in the validation logic.
* Minor documentation corrections and version/date bumps to reflect the new release. [[1]](diffhunk://#diff-9cc358405149db607ff830a16f0b4b21f7366e3c99ec00d52800acebe21b231cL4-R5) [[2]](diffhunk://#diff-c5312afd81ff117e4909140586797da8ebeb604732bd4aac2f65561bccc383a2L18-R18)

These changes improve the reliability and usability of metabolomics QC workflows and make the package easier to use and maintain.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved metabolomics internal-standard handling in name validation and acceptance of slash-delimited alternative names; validation now tolerates network/API issues without aborting.

* **Documentation**
  * Expanded README with overview, key features, installation, docs, and citation; updated changelog entry and docs to reflect default verbose behavior.

* **Tests**
  * Added tests covering validation edge cases, skipping of internal standards, empty inputs, and simulated API failures.

* **Chores**
  * Version bumped to 1.6.4.

#273

### CODE REVIEW

File: R/metabolomics_data_dictionary.R [hunk 1]
```diff
@@ -86,45 +86,74 @@ get_and_validate_mdd <- function(remove_duplications = FALSE, verbose = TRUE){
...
-    # Remove the standard label
-    standards <- "\\[.*\\]"
-    if(grepl(standards, rn)){
-      # message("\tStandard detected: ", rn, appendLF = FALSE)
-      rn <- gsub("(.*)(\\s+\\[iSTD\\])", "\\1", rn)
-      # message(" > ", rn)
     }
     
     search_api <- paste0("https://www.metabolomicsworkbench.org/rest/refmet/match/",URLencode(rn),"/name/")
     here <- jsonlite::fromJSON(search_api)
```
[the remark this answers is not in the document]
Comment (biodavidjm) [reply R/metabolomics_data_dictionary.R]: Not right. assumes the API returns "-" for Leucine/Isoleucine. It doesn’t. We verified empirically

File: R/metabolomics_data_dictionary.R [hunk 2]
```diff
@@ -86,45 +86,74 @@ get_and_validate_mdd <- function(remove_duplications = FALSE, verbose = TRUE){
...
-    }
-    # Remove the standard label
-    standards <- "\\[.*\\]"
-    if(grepl(standards, rn)){
-      # message("\tStandard detected: ", rn, appendLF = FALSE)
-      rn <- gsub("(.*)(\\s+\\[iSTD\\])", "\\1", rn)
-      # message(" > ", rn)
     }
     
     search_api <- paste0("https://www.metabolomicsworkbench.org/rest/refmet/match/",URLencode(rn),"/name/")
```
[the remark this answers is not in the document]
Comment (biodavidjm) [reply R/metabolomics_data_dictionary.R:144]: The Metabolomics Workbench REST router does not accept percent-encoded slashes here. Switching to reserved = TRUE would re-introduce the original bug from #273 in a different form (hard 404 instead of false RN2). Theoretically correct per RFC 3986, but practically wrong for this server.

File: R/metabolomics_data_dictionary.R [hunk 3]
```diff
@@ -86,45 +86,74 @@ get_and_validate_mdd <- function(remove_duplications = FALSE, verbose = TRUE){
...
+        if(grepl("/", rn, fixed = TRUE)){
+          rn_alternatives <- trimws(strsplit(rn, "/", fixed = TRUE)[[1]])
+          if(here$refmet_name %in% rn_alternatives){
+            next
+          }
+        }
+        if(verbose) message(paste0("   - (-) `refmet_name validation`: [`", rn, "`] must be modified to the RefMet Standarized name: \"", here$refmet_name, "\" (Error RN2)"))
         irm <- irm + 1
       }
     }
```
[the remark this answers is not in the document]
Comment (biodavidjm) [reply R/metabolomics_data_dictionary.R:173]: will be addressed

File: R/metabolomics_data_dictionary.R [hunk 4]
```diff
@@ -86,45 +86,74 @@ get_and_validate_mdd <- function(remove_duplications = FALSE, verbose = TRUE){
 
 #' @title Validate refmet_name
 #'
-#' @description Validate the refment_name using the Metabolomics Workbench API
+#' @description Validate the refmet_name using the Metabolomics Workbench API
 #' @param dataf (data.frame) Data frame with the refmet_name column
 #' @param verbose (logical) `TRUE` (default) shows messages)
 #' @return (numeric) number of refmet_name ids not available in RefMet
 #' @export
 validate_refmetname <- function(dataf, verbose){
```
[the remark this answers is not in the document]
Comment (biodavidjm) [reply R/metabolomics_data_dictionary.R]: Done!
